### PR TITLE
Refactor: Extract ui elements to components and update tailwind colors

### PR DESCRIPTION
**Pull request by jlndk**
* Make tailwind colors consistent with design system (and tailwind syntax)
* Extract button and checkbox to dedicated react components
    * Move custom css styles to said react components
* Replace png icons with their upstream font awesome counterparts (from their npm package)

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -4,22 +4,76 @@ import withMT from '@material-tailwind/react/utils/withMT';
...
+          200: '#DAD7FD',
+          300: '#AFA6FB',
+          400: '#9083FF',
+          500: '#7969FF',
+          600: '#604FED', // Base/600ID Primary bright purple
+          700: '#4F3ED8',
+          800: '#382AAD',
+          900: '#271D7B',
+        },
+        lightBlue: {
```
Comment (mickhansen) [inline tailwind.config.js]: I realize we also used camelCase before, but other projects use `light-blue` etc

File: src/components/AddToCartButton.tsx
Comment (jlndk) [inline src/components/AddToCartButton.tsx:1]: This wasn't used anywhere

File: src/components/QuantityControlButton.tsx
Comment (jlndk) [inline src/components/QuantityControlButton.tsx:1]: Here I consolidated the two cases (user having added the item to the cart or not), such that the plus button can be used for both cases

**Review by nmoskaleva — APPROVED** [review]
Looks good!

**Pull request merged**